File: app/src/main/java/com/liujing/pagerouter/example/MyFragment.java
package com.liujing.pagerouter.example;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.liujing.pagerouter.RouterArgsx;
import com.liujing.pagerouter.annotation.RouterField;
import com.liujing.pagerouter.annotation.RouterFragment;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

@RouterFragment(value = "myfragment", activityClazz = FragmentContainerActivity.class)
public class MyFragment extends Fragment {

    @RouterField("id")
    private int id;

    @RouterField("name")
    private String name;

    @RouterField("isShow")
    private boolean isShow;

    @RouterField("price")
    private float price;

    @RouterField("time")
    private long time;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.activity_common, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        RouterArgsx.inject(this);
        TextView textview = view.findViewById(R.id.params_text);
        String builder = getClass().getSimpleName() + '\n' +
                "id:" + String.valueOf(id) + '\n' +
                "name:" + String.valueOf(name) + '\n' +
                "isShow:" + String.valueOf(isShow) + '\n' +
                "price:" + String.valueOf(price) + '\n' +
                "time:" + String.valueOf(time) + '\n';
        textview.setText(builder);
    }
}

File: app/src/main/java/com/liujing/pagerouter/example/MyFragment.kt
package com.liujing.pagerouter.example

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.liujing.pagerouter.*
import com.liujing.pagerouter.annotation.RouterFragment

@RouterFragment(value = ["myfragment"], activityClazz = FragmentContainerActivity::class)
class MyFragment: Fragment() {

    private val mId: Int by routerIntArgOr("id")
    private val name: String by routerStringArgOr("name")
    private val price: Float by routerFloatArgOr("price")
    private val time: Long by routerLongArgOr("time")
    private val isShow: Boolean by routerBooleanArgOr("isShow", false)

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.activity_common, container, false)
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val textview = view.findViewById<TextView>(R.id.params_text)
        val builder = javaClass.simpleName + '\n'.toString() +
                "id:" + mId.toString() + '\n'.toString() +
                "name:" + name + '\n'.toString() +
                "isShow:" + isShow.toString() + '\n'.toString() +
                "price:" + price.toString() + '\n'.toString() +
                "time:" + time.toString() + '\n'.toString()
        textview.text = builder
    }
}